### PR TITLE
[RFC] Retrieve cpu and mem from grpc server

### PR DESCRIPTION
**Pull request by dpeng817**
This PR builds upon the metrics retrieval APIs established in https://github.com/dagster-io/dagster/pull/18721 to also include CPU and memory utilization. Assuming that we're in agreement that the method of cpu/mem retrieval established in https://github.com/dagster-io/dagster/pull/18816 is reasonable, then I think this PR is quite straightforward.

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18822?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #18721** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18721?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #18816** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18816?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #18822** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18822?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18822?utm_source=stack-comment).

File: python_modules/dagster/dagster/_grpc/server.py [hunk 1]
```diff
@@ -128,6 +129,19 @@ def _record_max_workers(max_workers: Optional[int]) -> None:
...
+    with _METRICS_LOCK:
+        last_cpu_measurement_time = _UTILIZATION_METRICS["general_info"].get(
+            "measurement_timestamp"
+        )
+        last_cpu_measurement = _UTILIZATION_METRICS["general_info"].get("cpu_time")
+        utilization_metrics = retrieve_containerized_utilization_metrics(
+            logger, last_cpu_measurement_time, last_cpu_measurement
+        )
+        for key, val in utilization_metrics.items():
+            _UTILIZATION_METRICS["general_info"][key] = val
```
Comment (rexledesma) [inline python_modules/dagster/dagster/_grpc/server.py:142]:
```suggestion
        _UTILIZATION_METRICS["general_info"].update(utilization_metrics)
```

File: python_modules/dagster/dagster/_grpc/server.py [hunk 2]
```diff
@@ -454,6 +468,8 @@ def ReloadCode(
 
     def Ping(self, request, _context: grpc.ServicerContext) -> api_pb2.PingReply:
         echo = request.echo
+        if self._enable_metrics:
+            _record_utilization_metrics(self._logger)
```
Comment (rexledesma) [inline python_modules/dagster/dagster/_grpc/server.py:472]: Can we centralize this in the retrieve_metrics decorator? Would prefer that decorator to be the source of truth for how `_UTILIZATION_METRICS` is mutated.

**Review by rexledesma — APPROVED** [review]
See comments

**Pull request merged**